### ISSUE TITLE
ppsspp/ext/glslang does not contain a CMakeLists.txt file.

### ISSUE THREAD

**Issue by Adrien2002** [hrydgard/ppsspp#9297]
I'm trying to build ppsspp on NetBSD amd64

$ cmake ..
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- No build type selected, default to Release
CMAKE_SYSTEM_PROCESSOR: x86_64
Generating for x86, Release
Generating for x86_64, Release
**CMake Error at ext/CMakeLists.txt:7 (add_subdirectory):**
  **The source directory**

   **/home/Adrien2002/Téléchargements/ppsspp/ext/glslang**

  **does not contain a CMakeLists.txt file.**

--  Found avcodec: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libavcodec.so.57.64.100
--  Found avdevice: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libavdevice.so.57.1.100
--  Found avfilter: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libavfilter.so.6.65.100
--  Found avformat: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libavformat.so.57.56.100
--  Found avutil: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libavutil.so.55.34.100
--  Found postproc: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libpostproc.so.54.1.100
--  Found swresample: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libswresample.so.2.3.100
--  Found swscale: /usr/pkg/include /usr/pkg/lib/ffmpeg3/libswscale.so.4.2.100
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/ui_atlas.zim
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/lang
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/4xhqglsl.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/4xhqglsl.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/5xBR.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/5xBR.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/aacolor.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/aacolor.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/bloom.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/cartoon.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/cartoon.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/crt.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/defaultshaders.ini
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/fxaa.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/fxaa.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/grayscale.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/inversecolors.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/natural.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/natural.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/scanlines.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/sharpen.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/upscale_spline36.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/upscale_spline36.vsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/shaders/vignette.fsh
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/ppge_atlas.zim
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/compat.ini
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/langregion.ini
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/gamecontrollerdb.txt
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/unknown.png
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/jpn0.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/kr0.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn0.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn1.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn10.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn11.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn12.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn13.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn14.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn15.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn2.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn3.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn4.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn5.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn6.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn7.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn8.pgf
-- Installing: /home/Adrien2002/Téléchargements/ppsspp/build/assets/flash0/font/ltn9.pgf
**-- Configuring incomplete, errors occurred!**
See also "/home/Adrien2002/Téléchargements/ppsspp/build/CMakeFiles/CMakeOutput.log".

Indeed, there isn't such file in ppsspp/ext/glslang folder. Why does it look into this folder for a CMakeLists.txt file ?

**Comment by Orphis** [comment]
Just like the million of other issues opened about this, you need to pull the submodules in the project.

**Comment by Bigpet** [comment]
since there's actually a cmakelists.txt in that repo it's more likely that he just didn't pull the submodules

**Comment by Rutcha** [comment]
I guess you must try 
`git submodule update --init --recursive
`
in order to populate those folders with some CMakeListsts.txt

**Comment by hrydgard** [comment]
I think you're about five and a half years late with that reply. Please don't dig up old closed threads like this when there's no new interesting info - this is even part of our build instructions.

**Comment by Rutcha** [comment]
I'm sorry, but you tell that to google:
I had this issue last night and that was still the first link in my search.
If you quote the error in the search, you get this link first, then you come here and you used to find no answear.
I didn't look for trouble, I looked up for a solution, got here, moved on, moved on again, then finally found out about submodules and after a while I came back to fix this thread or so I thought.
I never wanted to disturb the main ppsspp developer, I never imagined it.

p.s.
All praises to your work I could say are few, but ayway I must congratulate you. Thanks! Thank you very much
